File: 392-is-subsequence/is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        sIdx, tIdx = 0, 0

        while tIdx < len(t) and sIdx < len(s):
            if t[tIdx] == s[sIdx]:
                sIdx += 1
                tIdx += 1
            else:
                tIdx += 1
        return sIdx >= len(s)

File: 392-is-subsequence/is-subsequence.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isSubsequence = function(s, t) {
    let [tIdx, sIdx] = [0, 0];

    while (tIdx < t.length && sIdx < s.length){
        if (t[tIdx] === s[sIdx]){
            tIdx++;
            sIdx++;
        } else {
            tIdx++;
        }
    }

    return sIdx >= s.length;
};